### ISSUE TITLE
Always get SEHException with visual studio debug

### ISSUE THREAD

**Issue by realivanjx** [MelbourneDeveloper/Device.Net#109]
**Describe the issue**
I always got `System.Runtime.InteropServices.SEHException` if i run my app (test console app) with visual studio debugger. If i run without debugger, this error never appear. I suspect that it is thrown when im trying to dispose the `IDevice` object:

```
m_deviceService.Device.Dispose();
```

or even with this code:

```
await DeviceManager.Current.GetConnectedDeviceDefinitionsAsync(null);
```

Whereas `m_deviceService` is an instance of `BulkIOService`. I dont know which part of my code causes this error because visual studio says that it is executing external code (see screenshot below). I dont know if this is a real issue but it is quite annoying to always get it when debugging. Any idea?

**Screenshots**

![image](https://user-images.githubusercontent.com/12814796/63408991-743f2580-c41a-11e9-876f-548fdae9be26.png)

Platform: Windows 10
Version: 3.0.0 beta 2

**Comment by MelbourneDeveloper** [comment]
@realvanjx thanks for reporting. Yes, I've seen this one before. I'm not sure if it's a real issue myself...

Could you try including the device.net projects in your solution directly? Instead of the NuGet. I need to see the exact line.

**Comment by realivanjx** [comment]
I dont think that it will show the line either because i think it errors in the c++ api you are using.

**Comment by MelbourneDeveloper** [comment]
@realivanjx you're probably right. I had the same issue. Do you have "enable just my code" turned on? Check that it debugging options.

**Comment by realivanjx** [comment]
Yes it is on already

**Comment by MelbourneDeveloper** [comment]
@realivanjx the only thing I can recommend is to turn off the exception in the debugging options. I will look out for it and try to find out which API is causing the problem.

**Comment by realivanjx** [comment]
i have tried to disable the exception in debugging options but that doesnt work. the app also keeps closing when i hit continue on visual studio after the error. this makes me impossible to debug the app now

**Comment by realivanjx** [comment]
never got this issue when running without debugging though

**Comment by realivanjx** [comment]
is there any updates on this issue?

**Comment by MelbourneDeveloper** [comment]
@realivanjx I probably won't fix this issue because it doesnt cause any serious problems that I know about. 

It's only an annoyance. As far as I know, it only causes an issue for debugging right?

**Comment by realivanjx** [comment]
yea, but it affects me so that i cant debug the app at all now

**Comment by realivanjx** [comment]
so for me it kinda serious

**Comment by realivanjx** [comment]
maybe this could be the issue with visual studio itself? because this only happens while debugging though

**Comment by realivanjx** [comment]
![image](https://user-images.githubusercontent.com/12814796/69472586-a035c280-0dde-11ea-83e5-a56f2371ed0c.png)

**Comment by st-arnold** [comment]
Same problem here... probably it has something to do with SafeFileHandles, as VS was trying to locate source code for that (if Just My Code in debugger settings wasn't enabled).

**Comment by MelbourneDeveloper** [comment]
@st-arnold you're probably right. I'd prefer that everything were converted to safe handles if possible

**Comment by datoml** [comment]
I can confirm that this only happens during a debugger is attached.
You can easily reproduce the problem when you start for instance a reading process from a device and unplug the USB during that.
Reattach the USB and start again a reading process.
Then this SEHException occurs.
As said, when a debugger is attached to the app crashes.
Without one it running without any issue.

**Comment by MelbourneDeveloper** [comment]
@st-arnold @datoml @realivanjx 

This issue may have been because of an issue with dispose. Any chance you could try the latest version to see if you still get the issue?

https://www.nuget.org/packages/Usb.Net/4.0.8-alpha

**Comment by NorbiPeti** [comment]
I still have this issue using Usb.Net 4.0.9-alpha and Rider for debugging. The stack trace references SafeHandle.Finalize() (and none of my code). Disabling external code debugging seems to solve the issue.

**Comment by MelbourneDeveloper** [comment]
@NorbiPeti yep. No idea what this or how to fix it. Suggestions welcome.

**Comment by MelbourneDeveloper** [comment]
@NorbiPeti have you disabled "Just My Code" in visual studio debugging options?
